crates/services/src/services/git.rs

```diff
@@ -579,83 +579,120 @@ impl GitService {
         }
     }
 
-    /// Merge changes from a worktree branch back to the main repository
-    pub fn merge_changes(
+    /// Find where a branch is currently checked out
+    fn find_checkout_path_for_branch(
         &self,
         repo_path: &Path,
-        worktree_path: &Path,
         branch_name: &str,
-        base_branch_name: &str,
+    ) -> Result<Option<std::path::PathBuf>, GitServiceError> {
+        let git_cli = GitCli::new();
+        let worktree_list = git_cli
+            .git(repo_path, ["worktree", "list", "--porcelain"])
```

Comment:
Please add a method in `git_cli.rs` to list worktree and return a `Vec` of a struct with worktree metadata. I have added a doc-string to `fn git` to discourage from using it directly. I think I'll make it private, and create a shim for tests only under `#[cfg(test)`.